src/containers/TemplateBuilder/Templates.tsx

```diff
@@ -90,36 +105,83 @@ const ViewEditButton: React.FC<CellProps> = ({ template: { id } }) => {
   )
 }
 
-const ExportButton: React.FC<CellProps> = ({ template: { code, version, id } }) => {
-  const { exportTemplate } = useOperationState()
-  const snapshotName = `${code}-${version}`
+const ExportButton: React.FC<CellProps> = ({ template }) => {
+  const { exportTemplate, updateTemplate } = useOperationState()
+  const showToast = useToast({
+    style: 'success',
+    title: 'Template exported',
+    text: `${template.code} - ${getVersionString(template)}`,
+  })
   const JWT = localStorage.getItem(config.localStorageJWTKey)
+  const [open, setOpen] = useState(false)
+  const [commitMessage, setCommitMessage] = useState('')
+
+  const doExport = async (versionId = template.versionId) => {
+    const { code, versionHistory, id } = template
+    const snapshotName = `${code}-${versionId}_v${versionHistory.length + 1}`
+    if (await exportTemplate({ id, snapshotName })) {
+      const res = await fetch(
+        getServerUrl('snapshot', { action: 'download', name: snapshotName }),
+        {
+          headers: { Authorization: `Bearer ${JWT}` },
+        }
+      )
+      const data = await res.blob()
+      var a = document.createElement('a')
+      a.href = window.URL.createObjectURL(data)
+      a.download = `${snapshotName}.zip`
+      a.click()
+      // Delete the snapshot cos we don't want snapshots page cluttered
+      // with individual templates
+      await fetch(getServerUrl('snapshot', { action: 'delete', name: snapshotName }), {
+        method: 'POST',
+        headers: { Authorization: `Bearer ${JWT}` },
+      })
+      showToast()
+    } else showToast({ style: 'error', title: 'Problem exporting template' })
+  }
 
   return (
     <div key="export">
+      <Confirm
+        open={open}
+        // Prevent click in Input from closing modal
+        onClick={(e: any) => e.stopPropagation()}
+        content={
+          <div style={{ padding: 10, gap: 10 }} className="flex-column">
+            <h2>Commit and export template?</h2>
+            <p>
+              By exporting this template now, you will be committing the current version. To make
+              any further changes, you will need to duplicate it and start a new template version.
+            </p>
+            <div className="flex-row-start-center" style={{ gap: 10 }}>
+              <label>Please provide a commit message:</label>
+              <Input
```

Comment:
I'm getting an error when this commit message is left blank

Reply:
That's odd, I can't reproduce either of those errors. You might be getting conflicts due to using a snapshot that was created when reviewing #1541 (which you've approved), since that's a later branch than this one? 

Can you try with that branch and see if you still get this problem? Otherwise I'll get you to show me tomorrow.

Reply:
Sounds good - I'll have a play with alternative snapshots

Reply:
I'm still getting some issues with the new branch:

First when creating a new template version. After making this new version available, I can no longer edit despite no apparent constraints:
![1FC689CA-7720-499D-BDA0-C09ED1B8A417](https://github.com/openmsupply/conforma-web-app/assets/90807420/abcf1d85-2a1d-4164-824a-ce72baf19338)

Secondly if I try to make a new version based on this uncommitted version:

![DB437597-012F-4124-994B-6DED984CCE9E](https://github.com/openmsupply/conforma-web-app/assets/90807420/a639bfb5-06e3-4244-b107-66d8936228bb)

This is from the Fiji live server snapshot.